llvm/lib/Target/AIE/AIECombinerHelper.cpp

```diff
@@ -63,11 +63,11 @@ MachineInstr *findPreIncMatch(MachineInstr &MemI, MachineRegisterInfo &MRI,
   return nullptr;
 }
 
-/// Checks if any operand of \a Use is defined by \a MI.
-/// This is not transitive: it will not look at how the uses of \a MI are
+/// Checks if any operand of \a MI is defined by \a Def.
+/// This is not transitive: it will not look at how the uses of \a Def are
 /// defined.
-bool isUseOf(const MachineInstr &MI, const MachineInstr &Use) {
-  for (auto &Defs : Use.defs()) {
+bool isUseOf(const MachineInstr &MI, const MachineInstr &Def) {
+  for (auto &Defs : Def.defs()) {
```

Comment:
Does that iterate on implicit defs as well? I'm never sure

Reply:
No, just explicit ones! To cover all definitions, we need `all_defs()`

Reply:
I guess we should be careful about implicit defs as well. The instruction might write to a control register that is used by a later instruction. I don't think that case can happen in AIE2, but who knows.

However, I'm now thinking about a case like this:
```
%0:vec256 = VLD
$crSat = ...
%1:vec512 = VUNPACK %0, implicit $crSat
```
There we cannot advance the VUNPACK, because it would then use a different value for `$crSat`.

Reply:
Yes, this could be a case if we extend for VUPS case...

Reply:
The best, in this case, is to iterate over all uses and defs, like:

```
bool isUseOf(const MachineInstr &MI, const MachineInstr &Def) {
  for (auto &Defs : Def.all_defs()) {
    for (auto &MIUse : MI.all_uses()) {
      if (MIUse.isReg() && Defs.getReg() == MIUse.getReg())
        return true;
    }
  }
  return false;
}
```

Reply:
Oh alright, it's VPACK that uses `crSat`, not VUNPACK. Still, the latter could use `crUnpackSign`, so we can end up with the same problem. I think we need to use `all_uses` and `all_defs` like you proposed.